### PR TITLE
Add CSIDriver creation factory to e2e framework

### PR DESCRIPTION
**Pull request by wongma7**
**What type of PR is this?**
/kind feature

**What this PR does / why we need it**: Now that we expect drivers to include these with their install manifests, this will be useful: https://github.com/kubernetes-csi/docs/pull/183/files

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

**Comment by wongma7** [comment]
/sig storage
/cc @pohly 
/cc @jsafrane 
thanks all.

File: test/e2e/framework/create.go [hunk 1]
```diff
@@ -286,6 +287,7 @@ var factories = map[What]ItemFactory{
 	{"ServiceAccount"}:     &serviceAccountFactory{},
 	{"StatefulSet"}:        &statefulSetFactory{},
 	{"StorageClass"}:       &storageClassFactory{},
+	{"CSIDriver"}:          &csiDriverFactory{},
```
Comment (cwdsuzhou) [inline test/e2e/framework/create.go]: Can we add it in alphabetical order?

**Review by pohly — CHANGES_REQUESTED** [review]
Makes sense, but as @cwdsuzhou said, please keep the `factories` map sorted by name.

File: test/e2e/framework/create.go (hunk 1 above)
Comment (wongma7) [reply test/e2e/framework/create.go]: fixed

**Review by pohly — APPROVED** [review]
/lgtm
/approve

**Pull request merged**